### PR TITLE
Processing algorithm help

### PR DESCRIPTION
**Pull request by ghtmtt**
Second PR for improving Processing helping system for QGIS Geometry tool. 

I started a *new* page with just the available algorithms right now (master compiled). When other algorithms are added, I'll add them too in the correct section.

I think it better to proceed like this so we can have updated docs for the algorithms and we can discuss on how to edit the page etc..

If someone think it is better that I work only on local machine and make a single PR when we are sure all the algorithms are added let me know and I'll close the PR without merging it.

**BTW**: the buffer algorithm is not documented because some problems arise when using it.

### CODE REVIEW

**Comment by DelazJ** [comment]
Matteo, I'd say "add PRs when you are ready". Better do it that way than waiting to have all the algorithms ready in application before pulling requests (nobody knows when Nyall will finish his refactoring and you'd then be in a hurry and reviewers also). Doing it by small bits also give time to really review and have a good and tested template (and avoid reorganization at the deadline).

For algorithms, maybe you can have a list of existing ones in 2.18 (or just before Armageddon begins) and check them as soon as you document. It can be an issue report, hence give others an overview of what's done and what's remaining... (will serve like a report for the grant proposal btw)

**Comment by ghtmtt** [comment]
@DelazJ good that also you agree to that! You get the point: doing it step by step avoids me to be in a hurry when all he algorithms are back ported and also reviewers.. 

Moreover, having a first draft of what the docs will look like is also a good chance to have some re-styling.

Thanks for the feedback.

File: source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst [hunk 1]
```diff
@@ -31,331 +32,299 @@ Parameters
...
 ......
 
-``Output layer`` [vector: point, line]
+``Boundary`` [vector: point, line]
   Boundary from the input layer (point for line, and line for polygon).
 
+.. figure:: /static/user_manual/processing_algs/qgis/boundary_lines.png
+   :align: center
+
+   Boundary layer for lines. In yellow one selected features
```
Comment (SrNetoChan) [inline source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst]: `Boundary of lines layer`, maybe?

(...)one selected feature. (remove s)

File: source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst [hunk 2]
```diff
@@ -13,14 +13,15 @@ Vector geometry
...
-Description
-...........
+For **polygon geometries** , the boundary consists of all the line strings for
+each ring of the polygon.
 
-Returns the closure of the combinatorial boundary of the input geometries (ie
-the topological boundary of the geometry). For instance, a polygon geometry
-will have a boundary consisting of the linestrings for each ring in the
-polygon. Only valid for polygon or line layers.
+For **lines geometries**, the boundaries are the nodes between each features.
```
Comment (SrNetoChan) [inline source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst]: each feature

File: source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst [hunk 3]
```diff
@@ -31,331 +32,299 @@ Parameters
...
+.. figure:: /static/user_manual/processing_algs/qgis/boundary_lines.png
+   :align: center
+
+   Boundary layer for lines. In yellow one selected features
+
+.. figure:: /static/user_manual/processing_algs/qgis/boundary_polygon.png
+   :align: center
+
+   Boundary for polygon. In light blue the polygon and in dark blue the boundary
+
```
Comment (SrNetoChan) [inline source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst]: try being consistent with the legende used for lines. either add 'layer' in both, or remove from lines.

Can you move this example on top of the lines one? just to keep the order used in the description.

File: source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst [hunk 4]
```diff
@@ -31,331 +32,299 @@ Parameters
...
 
-``Valid output`` [vector]
-  Output valid features, unchanged.
-
-``Invalid output`` [vector]
-  Output invalid features, with an additional *_errors* field describing the validity problems.
-
-``Error output`` [vector]
-  Output exact position on the validity problems as a point layer with a *message* field.
+``Bounds`` [vector: poylgon]
```
Comment (SrNetoChan) [inline source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst]: polygon

File: source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst [hunk 5]
```diff
@@ -31,331 +32,299 @@ Parameters
...
 
-See also
-........
 
-Concave hull
-------------
+.. _qgis_buffer:
+
+Buffer
+------
```
Comment (SrNetoChan) [inline source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst]: empty line is missing after this.

File: source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst [hunk 6]
```diff
@@ -31,331 +32,299 @@ Parameters
...
+This algorithm computes a buffer area for all the features in an input layer,
+using a fixed or dynamic distance.
+
+The segments parameter controls the number of line segments to use to approximate
```
Comment (SrNetoChan) [inline source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst]: can we use bold or italic to highlight parameters names?

File: source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst [hunk 7]
```diff
@@ -31,331 +32,299 @@ Parameters
...
+ a quarter circle when creating rounded offsets.
```
Comment (SrNetoChan) [inline source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst]: Formatting should be consistent. either insert a new line for each parameter or don't.

File: source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst [hunk 8]
```diff
@@ -31,331 +32,299 @@ Parameters
...
 
-Description
-...........
+The end cap style parameter controls how line endings are handled in the buffer.
+The join style parameter specifies whether round, mitre or beveled joins should
+be used when offsetting corners in a line.
 
-<put algorithm description here>
+The mitre limit parameter is only applicable for mitre join styles, and controls
+the maximum distance from the offset curve to use when creating a mitred join.
```
Comment (SrNetoChan) [inline source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst]: Why not describe all this above in the parameters section?

File: source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst [hunk 9]
```diff
@@ -31,331 +32,299 @@ Parameters
...
-........
 
-Convert geometry type
----------------------
+.. _qgis_centroids:
+
+Centroids
+---------
+This algorithm creates a new point layer, with points representing the centroid
+of the geometries of the inpu layer.
```
Comment (SrNetoChan) [inline source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst]: input

File: source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst [hunk 10]
```diff
@@ -31,331 +32,299 @@ Parameters
...
+``Centroids`` [vector: point]
+  Points vector layer in output.
 
 Console usage
 .............
 
 ::
 
-  processing.runalg('qgis:convertgeometrytype', input, type, output)
+  processing.runalg('qgis:fixeddistancebuffer', input, distance, segments, dissolve, output)
```
Comment (SrNetoChan) [inline source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst]: ?centroids?

File: source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst [hunk 11]
```diff
@@ -31,331 +32,299 @@ Parameters
...
+information:
 
-Description
-...........
+.. figure:: /static/user_manual/processing_algs/qgis/check_validity.png
+   :align: center
 
-<put algorithm description here>
+   Left the input layer. Right: in blue the valid layer, in green the invalid layer
+   and the red point identifies the exact error coordinates.
```
Comment (SrNetoChan) [inline source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst]: Consistency, either use `:` to separate Left and right, or don't. I would put them in brackets instead.

**Comment by DelazJ** [comment]
see #2240 babies

**Pull request closed without merging**